### ISSUE TITLE
Service Worker not caching anything

### ISSUE THREAD

**Issue by GaryB432** [angular/angular-cli#10361]
### Versions
```
Angular CLI: 6.0.0-rc.4
Node: 9.3.0
OS: win32 x64
Angular: 6.0.0-rc.5
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router, service-worker

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.5.6
@angular-devkit/build-angular     0.5.6
@angular-devkit/build-optimizer   0.5.6
@angular-devkit/core              0.5.6
@angular-devkit/schematics        0.5.6
@angular/cli                      6.0.0-rc.4
@ngtools/json-schema              1.1.0
@ngtools/webpack                  6.0.0-rc.4
@schematics/angular               0.5.6
@schematics/update                0.5.6
rxjs                              6.0.0-uncanny-rc.7
typescript                        2.7.2
webpack                           4.5.0
```

### Repro steps
```  
git clone https://github.com/GaryB432/angular-pwa
cd angular-pwa
git fetch origin
git checkout -b test origin/v6
npm install
ng build --prod
docker build -t issue-repro .
docker run -d -p 8080:80 issue-repro
```
visit http://localhost:8080 and reload/clear until the sw registers

### Observed behavior

Chrome Dev Tools shows
- IndexedDB is empty
- many caches but all empty save one

### Desired behavior
I expected `IndexedDB` and `Cache Storage` to contain some entries like in 1.7

### Mention any other details that might be useful (optional)

I notice `ngsw.json` is quite different but I believe it used to contain the hashed names.

- why so many caches? What is their purpose?
- How to make use of `safety-worker`?  I want to build a 'kill-switch' and I suppose that is safety-worker's purpose. Is a kill-switch already built-into `ngsw.worker.js`?

**Comment by filipesilva** [comment]
@GaryB432 can you try `@angular/cli@6.0.0-rc.5` together with `@angular-devkit/build-angular@0.5.7` please? 

I think I fixed this with https://github.com/angular/devkit/pull/709 but there might be more leftover cases.

**Comment by GaryB432** [comment]
Thanks for looking. I will get to this soon since I can only work on this cool stuff in my spare time. Looking at your related merge I see '@angular/pwa'. Should I be using that instead of rolling SW by hand? See https://stackoverflow.com/questions/49880891/how-to-use-the-angular-pwa-schematics-library

**Comment by GaryB432** [comment]
The repo above is now using `@angular/cli@6.0.0-rc.5` and `@angular-devkit/build-angular@0.5.7`. I also used `ng add @angular/pwa` instead of adding sw by hand now that I know about `add`.  It does look like you nailed `ngsw.json`. 

However

- IndexedDB and caches still unexpectedly empty
- `manifest.json` is under `assets` instead of in the root.  I can open an issue with `@angular\pwa` if you need that

**Comment by gkalpak** [comment]
The `manifest.json` issues have been fixed in #722. I tried with `@angular/cli@6.0.0-rc.7` and everything (mentioned in this issue) seems to work fine. IndexedDB is empty (as expected) and the caches are populated.

Closing. Please, open a new issue if there are still problems/things to be improved.
